oap-server/server-core/src/test/java/org/apache/skywalking/oap/server/core/query/TopologyQueryServiceTest.java

```diff
@@ -0,0 +1,124 @@
+    public void setUp() throws Exception {
+        when(moduleServiceHolder.getService(ITopologyQueryDAO.class)).thenReturn(queryDAO);
+    }
+
+    @Test
+    public void getGlobalTopology() throws Exception {
+        List<Call.CallDetail> serverCalls = mockServerCall();
+        List<Call.CallDetail> clientCalls = mockClientCall();
+
+        when(queryDAO.loadServerSideServiceRelations(Step.MONTH, START_TB, END_TB)).thenReturn(serverCalls);
+        when(queryDAO.loadClientSideServiceRelations(Step.MONTH, START_TB, END_TB)).thenReturn(clientCalls);
+        Topology topology = queryService.getGlobalTopology(Step.MONTH, START_TB, END_TB, 123, 234);
+        assertNotNull(topology);
+        assertTopology(topology);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void getServiceTopology() throws Exception {
+
+        List<Call.CallDetail> serverCalls = mockServerCall();
+        List<Call.CallDetail> clientCalls = mockClientCall();
+
+        List<Call.CallDetail> sourceCalls = mockSourceServiceCall();
+
+        when(queryDAO.loadSpecifiedServerSideServiceRelations(any(Step.class), anyLong(),
+                anyLong(), anyListOf(Integer.class))).thenReturn(serverCalls, sourceCalls);
+        when(queryDAO.loadSpecifiedClientSideServiceRelations(any(Step.class), anyLong(),
+                anyLong(), anyListOf(Integer.class))).thenReturn(clientCalls);
+
+        Topology topology = queryService.getServiceTopology(Step.MONTH, START_TB, END_TB, 12);
+        assertTopology(topology);
+        assertTrue(topology.getNodes().stream()
+                .anyMatch(node -> "component-name-3".equals(node.getType()) && node.getId() == 33));
+    }
+
+    @Test
+    public void getEndpointTopology() throws Exception {
+        List<Call.CallDetail> serverCalls = mockServerCall();
+        when(queryDAO.loadSpecifiedDestOfServerSideEndpointRelations(any(Step.class), anyLong(),
+                anyLong(), anyInt())).thenReturn(serverCalls);
+        Topology topology = queryService.getEndpointTopology(Step.MONTH, START_TB, END_TB, 1);
+
+        assertNotNull(topology);
+
+        assertEquals(4, topology.getCalls().size());
+
+        for (int i = 0; i < 4; i++) {
+            Call.CallDetail serverDetail = serverCalls.get(i);
+            Call topologyCall = topology.getCalls().get(i);
+            assertEquals(serverDetail.getSource(), topologyCall.getSource());
+            assertEquals(serverDetail.getTarget(), topologyCall.getTarget());
+        }
+
+        assertEquals(8, topology.getNodes().size());
+        assertNode(0, topology.getNodes().get(0));
+        assertNode(100, topology.getNodes().get(1));
+        assertNode(101, topology.getNodes().get(2));
+        assertNode(111, topology.getNodes().get(3));
+        assertNode(102, topology.getNodes().get(4));
+        assertNode(122, topology.getNodes().get(5));
+        assertNode(103, topology.getNodes().get(6));
+        assertNode(133, topology.getNodes().get(7));
```

Comment:
`assertNode` is very hard to understand. Please use logic name to replace numbers.